src/serac/physics/solid_mechanics.hpp

```diff
@@ -801,6 +801,55 @@ class SolidMechanics<order, dim, Parameters<parameter_space...>, std::integer_se
                                  mesh_, qdata);
   }
 
+  /**
+   * @brief Functor representing a material stress.  A functor is used here instead of an
+   * extended, generic lambda for compatibility with NVCC.
+   */
+  template <typename Material>
+  struct MaterialStressFunctor {
+    /// @brief Constructor for the functor
+    MaterialStressFunctor(Material material, GeometricNonlinearities gn) : material_(material), geom_nonlin_(gn) {}
```

Comment:
Why not take material by const lvalue reference? Similarly for gn? That way there's only one copy for each instead of two.

Reply:
Or are these very small objects that would be better to copy by value than passing around references?

Reply:
I'll just go ahead and just remove all the unnecessary forwarding references in this PR

Reply:
just chiming in to say that I agree with moving away from forwarding and using const references instead

Reply:
for reference, this is a useful library for tracking which kinds of constructors are called and how many times:
https://github.com/VincentZalzal/noisy

with an example here:
https://godbolt.org/z/v974sMEzx